x/ibc/04-channel/keeper/packet.go

```diff
@@ -290,66 +281,68 @@ func (k Keeper) RecvPacket(
 // and acted upon.
 func (k Keeper) AcknowledgePacket(
 	ctx sdk.Context,
-	packet exported.PacketI,
-	acknowledgement []byte,
+	packet types.Packet,
+	acknowledgement types.PacketDataI,
 	proof commitment.ProofI,
 	proofHeight uint64,
-	portCapability sdk.CapabilityKey,
-) (exported.PacketI, error) {
+) (types.Packet, error) {
 	channel, found := k.GetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel())
 	if !found {
-		return nil, types.ErrChannelNotFound(k.codespace, packet.GetSourcePort(), packet.GetSourceChannel())
+		return types.Packet{}, types.ErrChannelNotFound(k.codespace, packet.GetSourcePort(), packet.GetSourceChannel())
 	}
 
 	if channel.State != types.OPEN {
-		return nil, types.ErrInvalidChannelState(
+		return types.Packet{}, types.ErrInvalidChannelState(
 			k.codespace,
 			fmt.Sprintf("channel state is not OPEN (got %s)", channel.State.String()),
 		)
 	}
 
-	if !k.portKeeper.Authenticate(portCapability, packet.GetSourcePort()) {
-		return nil, errors.New("invalid capability key")
-	}
+	// XXX
```

Comment:
Why is this being changed in this PR?

Reply:
RecvPacket/AcknowledgePacket are called by the antehandler, which will route the messages/packets as permitted by the router initialization